### PR TITLE
[Shader Management] Foundry Multiple Shader Output - Graphics Repo

### PR DESCRIPTION
**Pull request by cdxntchou**
Repos:
Unity: https://github.cds.internal.unity3d.com/unity/unity/pull/7946
Graphics: https://github.com/Unity-Technologies/Graphics/pull/7056
Prototype: https://github.cds.internal.unity3d.com/joshua-davis/BlockSurfaceShaders/pull/1

OLD PRs (for reference):
Graphics: https://github.com/Unity-Technologies/Graphics/pull/6824
Unity: https://github.cds.internal.unity3d.com/unity/unity/pull/7609

---
### Purpose of this PR
Add support for multiple shader output to Shader Foundry

---
### Testing status
Ran existing foundry tests locally

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.shadergraph/Editor/ShaderFoundry/Samples/SimpleSample1.cs [hunk 1]
```diff
@@ -12,13 +12,14 @@ internal static void RunSimpleTest()
...
             var target = SimpleSampleBuilder.GetTarget();
+
             var container = new ShaderContainer();
-            var shaderBuilder = new ShaderBuilder();
             SimpleSampleBuilder.BuildCommonTypes(container);
-            SimpleSampleBuilder.Build(container, target, ShaderName, BuildSample, shaderBuilder);
 
-            var code = shaderBuilder.ToString();
-            DisplayTestResult(ShaderName, code);
+            var primaryShaderID = string.Empty;
```
Comment (jessebarker) [inline com.unity.shadergraph/Editor/ShaderFoundry/Samples/SimpleSample1.cs:19]: It looks like the Build call has the final parameter default initialized.  Do you need this empty string, or would the effect be the same if you didn't supply the parameter?

Comment (cdxntchou) [reply com.unity.shadergraph/Editor/ShaderFoundry/Samples/SimpleSample1.cs:19]: Yes, it should be equivalent to not supplying the parameter.   I was adding it to show you could supply it (as this is a bit of sample code).. and to call out that it is only generating the primary shader here.

**Review by jessebarker — APPROVED** [review]
LGTM

File: com.unity.shadergraph/Editor/ShaderFoundry/Samples/SimpleSample1.cs (hunk 1 above)
Comment (jessebarker) [reply com.unity.shadergraph/Editor/ShaderFoundry/Samples/SimpleSample1.cs:19]: Cool.  Just checking.

**Review by joshua-davis — APPROVED** [review]

File: com.unity.shadergraph/Editor/ShaderFoundry/Samples/SimpleSampleBuilder.cs [hunk 2]
```diff
@@ -8,20 +9,50 @@ internal static class SimpleSampleBuilder
...
+            {
+                if (!foundShaderIDs.Contains(template.AdditionalShaderID))
+                {
+                    yield return template.AdditionalShaderID;
+                    foundShaderIDs.Add(template.AdditionalShaderID);
+                }
+            }
+        }
+
+        // builds a specific shader ID (null builds the primary shader) on the gi
```
Comment (joshua-davis) [inline com.unity.shadergraph/Editor/ShaderFoundry/Samples/SimpleSampleBuilder.cs:37]: nit: "on the gi"? What is "gi"?

**Pull request merged**